cli/src/template_parser.rs

```diff
@@ -1019,6 +1019,36 @@ mod tests {
             parse_into_kind(r#" '"' "#),
             Ok(ExpressionKind::String(r#"""#.to_owned())),
         );
+
+        // Hex bytes
+        assert_eq!(
+            parse_into_kind(r#""\x61\x65\x69\x6f\x75""#),
+            Ok(ExpressionKind::String("aeiou".to_owned())),
+        );
+        assert_eq!(
+            parse_into_kind(r#""\xe0\xe8\xec\xf0\xf9""#),
+            Ok(ExpressionKind::String("àèìðù".to_owned())),
+        );
+        assert_eq!(
+            parse_into_kind(r#""\x""#),
+            Err(TemplateParseErrorKind::SyntaxError),
+        );
+        assert_eq!(
+            parse_into_kind(r#""\xf""#),
+            Err(TemplateParseErrorKind::SyntaxError),
+        );
+        assert_eq!(
+            parse_into_kind(r#""\xgg""#),
+            Err(TemplateParseErrorKind::SyntaxError),
+        );
```

Comment:
nit: can you copy hex bytes tests to `test_parse_string_literal()` in revset/fileset_parser.rs?

I don't think extensive test for `\e` is needed there, but maybe we can add `\e` to `"\<char>" escapes` test.